src/System.Collections/tests/Resources/System.Collections.Tests.rd.xml

```diff
@@ -1,7 +1,15 @@
+<?xml version="1.0" encoding="utf-8" ?>
 <Directives xmlns="http://schemas.microsoft.com/netfx/2013/01/metadata">
-  <Library>
+  <Library Name="System.Collections.Tests">
+    <Assembly Name="*Application*" DoNotInline="true" DoNotOptimize="true" />
```

Comment:
> Removing these causes some tests to fail (not serialization tests).

@safern Where do we track the codegen bug? I can't find any mention of it in the pull request that added it.

I did a sweep of the codebase for unaccounted codegen workarounds (e.g. #22475) but I forgot about this sneaky trick. We really should stick to the `<ILCBuildType Condition="'$(TargetGroup)' == 'uap'">chk</ILCBuildType>` pattern for this.